### PR TITLE
Remove public ExternalPluginValidationPlugin

### LINKED ISSUE #21792: De-incubate or remove ExternalPluginValidationPlugin
the `@Incubating` type `org.gradle.plugin.devel.plugins.ExternalPluginValidationPlugin` is part of the public API, but is not documented. It is unclear how this plugin is useful to users, but it is used by some external plugins currently.

We should either de-incubate and document it, or move it to a Gradle build-only scope.

### PR DESCRIPTION
**Pull request by lptr**
Fixes #21792

This plugin is only intended to be used in smoke tests, so now it lives in an init script there.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle help

**Comment by lptr** [comment]
@bot-gradle test AST please.

File: subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/ExternalPluginValidationPlugin.java [hunk 1]
```diff
@@ -1,166 +0,0 @@
...
- *
- * See the user guide for more information on
- * <a href="https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:up_to_date_checks" target="_top">incremental build</a> and
- * <a href="https://docs.gradle.org/current/userguide/build_cache.html#sec:task_output_caching" target="_top">caching task outputs</a>.
- *
- * @since 7.0
- */
-@Incubating
-@CacheableTask
-public abstract class ExternalPluginValidationPlugin implements Plugin<Project> {
```
Comment (asodja) [inline subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/ExternalPluginValidationPlugin.java:62]: :question: Will we add also a note in the breaking changes docs, to not surprise anyone that might accidentally use that?

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/WithPluginValidation.groovy [hunk 2]
```diff
@@ -73,15 +73,23 @@ trait WithPluginValidation {
         }
 
         protected void performValidation(List<String> extraParameters = []) {
+            owner.file("validate-external-gradle-plugin.gradle.kts") << getClass().getResource("validate-external-gradle-plugin.gradle.kts").text
+
             def failsValidation = validations.any { !it.messages.isEmpty() }
-            def validation = owner.runner(["validateExternalPlugins", "--continue", *extraParameters] as String[])
+            def validation = owner.runner([
+                "--init-script", "validate-external-gradle-plugin.gradle.kts",
```
Comment (asodja) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/WithPluginValidation.groovy:80]: ❔ Why did you decide to go with the init script? Would it be more difficult to keep it as a plugin?

**Review by asodja — APPROVED** [review]
LGTM overall, I think it would be good to add a note in the breaking changes.

File: subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/ExternalPluginValidationPlugin.java (hunk 1 above)
Comment (lptr) [reply subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/ExternalPluginValidationPlugin.java:62]: Added.

**Review by asodja — APPROVED** [review]
LGTM

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/WithPluginValidation.groovy (hunk 2 above)
Comment (asodja) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/WithPluginValidation.groovy:80]: As discussed on Slack current solution is simpler.

**Pull request merged**